source/client/process_impl.cc

```diff
@@ -1029,7 +1045,7 @@ bool ProcessImpl::run(OutputCollector& collector) {
   }
 
   try {
-    return runInternal(collector, tracing_uri, dns_resolver, options_.scheduled_start());
+    return runInternal(collector, tracing_uri, dns_resolver.value(), options_.scheduled_start());
```

Comment:
(nit) With StatusOr it is preferred to dereference (`*dns_resolver`) instead of using `.value()`.